### PR TITLE
fix/ `HangingOrderTracker` prematurely considering active orders as hanging orders

### PR DESCRIPTION
**Pull request by CrimsonJacket**
**Before submitting this PR, please make sure**:

- [x] Your code builds clean without any errors or warnings
- [x] You are using approved title ("feat/", "fix/", "docs/", "refactor/")

**A description of the changes proposed in the pull request**:

It appears the `HangingOrderTracker` is considering orders to be hanging orders prematurely.
An active order is considered to be a hanging order iff the corresponding order pair has been filled.

Blocks #3268

### CODE REVIEW

File: hummingbot/strategy/hanging_orders_tracker.py [hunk 1]
```diff
@@ -159,10 +164,15 @@ def is_hanging_order_in_strategy_active_orders(self, order: HangingOrder) -> boo
                        order.amount == o.quantity) for o in self.strategy.active_orders)
 
     def is_order_to_be_added_to_hanging_orders(self, order: LimitOrder) -> bool:
-        hanging_order = self._get_hanging_order_from_limit_order(order)
-        if hanging_order in self.equivalent_orders.union(self.orders_to_be_created):
-            return any(o.order_id == order.client_order_id
-                       for o in self.equivalent_orders.union(self.orders_to_be_created))
+        # Order will be added to hanging orders iff the corresponding order pair has been completely filled
```
Comment (CrimsonJacket) [inline hummingbot/strategy/hanging_orders_tracker.py:167]: Currently `HangingOrderTracker` does not handle partially filled orders. We will be addressing this in another PR.

**Review by aarmoa — CHANGES_REQUESTED** [review]
I have found that the issue is not in the business logic, but in the way the failing test perform the steps. In the test the orders are generated before the strategy is started, and thus the orders are considered as pre-existing by the strategy during startup (and added as hanging orders).
The change in this PR is not required.

**Comment by aarmoa** [comment]
Closing this PR after discussing it with @CrimsonJacket and verifying it is not required
